libtiledbsoma/src/utils/arrow_adapter.cc

```diff
@@ -443,4 +540,51 @@ std::string_view ArrowAdapter::to_arrow_format(
         tiledb::impl::type_to_str(datatype)));
 }
 
+// FIXME: Add more types, maybe make it a map
+enum ArrowType ArrowAdapter::to_nanoarrow_type(std::string_view sv) {
+    if (sv == "i")
+        return NANOARROW_TYPE_INT32;
+    else if (sv == "c")
+        return NANOARROW_TYPE_INT8;
+    else if (sv == "C")
+        return NANOARROW_TYPE_UINT8;
+    else if (sv == "s")
+        return NANOARROW_TYPE_INT16;
+    else if (sv == "S")
+        return NANOARROW_TYPE_UINT16;
+    else if (sv == "I")
+        return NANOARROW_TYPE_UINT32;
+    else if (sv == "l")
+        return NANOARROW_TYPE_INT64;
+    else if (sv == "L")
+        return NANOARROW_TYPE_UINT64;
+    else if (sv == "f")
+        return NANOARROW_TYPE_FLOAT;
+    else if (sv == "g")
+        return NANOARROW_TYPE_DOUBLE;
+    else if (sv == "u")
+        return NANOARROW_TYPE_STRING;
+    else if (sv == "U")
+        return NANOARROW_TYPE_LARGE_STRING;
+    else if (sv == "b")
+        return NANOARROW_TYPE_BOOL;
+    else if (sv == "tss:")
+        return NANOARROW_TYPE_INT64;  // NB time resolution set indepedently
+    else if (sv == "tsm:")
+        return NANOARROW_TYPE_INT64;  // NB time resolution set indepedently
+    else if (sv == "tsn:")
+        return NANOARROW_TYPE_INT64;  // NB time resolution set indepedently
+    else if (sv == "tsu:")
+        return NANOARROW_TYPE_INT64;  // NB time resolution set indepedently
+    else if (sv == "tdD")
+        return NANOARROW_TYPE_INT32;  // R Date: fractional days since epoch
+    else if (sv == "z")
+        return NANOARROW_TYPE_BINARY;
+    else if (sv == "Z")
+        return NANOARROW_TYPE_LARGE_BINARY;
+    else
+        throw TileDBSOMAError(fmt::format(
+            "ArrowAdapter: Unsupported TileDB datatype string: {} ", sv));
```

Comment:
Nanoarrow format strings here, not TileDB

Reply:
Actually Arrow format string (which is the same as Arrow).  Will adjust.